lib/validator.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Validator do
+  @moduledoc """
+  The module that handles checking to see if the user input is numerical for a Tic-Tac-Toe game
+  """
+
+  @type message :: :non_numerical
+
+  @spec validate(String.t()) :: {atom(), non_neg_integer() | message}
+  def validate(user_input) when is_binary(user_input) do
```

Comment:
Not suggesting a change, just food for thought: I imagine we'll be adding more validation rules soon and the current implementation requires us to update the validate method every time we add a rule. Is there a way we can make `validate` composable?

Reply:
🤔 Hmmm, that is a good point, we would have to also update the `@type` everytime. This is good thing to look into, Ill try to see how it evolves and how I can change it after some more validations are added, thank you! 😄